torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -368,11 +368,23 @@ class TORCH_API ProcessGroupNCCL : public Backend {
   // against a different set of devices, the process group creates another NCCL
   // communicator. These NCCL communicators are cached and reused if possible.
   //
+  ProcessGroupNCCL(
+      DistributedBackendOptions& opts,
+      c10::intrusive_ptr<Options> options = Options::create());
+
   ProcessGroupNCCL(
       const c10::intrusive_ptr<Store>& store,
       int rank,
       int size,
-      c10::intrusive_ptr<Options> options = Options::create());
+      c10::intrusive_ptr<Options> options = Options::create())
+  : Backend(rank, size) {
+    DistributedBackendOptions opts;
+    opts.store = store;
+    opts.group_rank = rank;
+    opts.group_size = size;
+    /* leave default values for the remaining opts parameters */
+    ProcessGroupNCCL(opts, options);
```

Comment:
```suggestion
    ProcessGroupNCCL(opts, std::move(options));
```
Doesn't c10::instrusive_ptr have a move ctor?